crates/autopilot/src/solvable_orders.rs

```diff
@@ -498,13 +489,53 @@ fn get_orders_with_native_prices(
         }
     });
 
+    let high_priority_tokens = prioritize_missing_prices(filtered);
+    native_price_estimator.replace_high_priority(high_priority_tokens);
+
     // Record separate metrics just for missing native token prices for market
     // orders, as they should be prioritized.
     metrics
         .auction_market_order_missing_price
         .set(filtered_market_orders);
 
-    (orders, used_prices)
+    (usable, used_prices)
+}
+
+/// Computes which missing native prices are the most urgent to fetch.
+/// Prices for recent orders have the highest priority because those are most
+/// likely market orders which users expect to get settled ASAP.
+/// For the remaining orders we prioritize token prices that are needed the most
+/// often. That way we have the chance to make a majority of orders solvable
+/// with very few fetch requests.
+fn prioritize_missing_prices(mut orders: Vec<Order>) -> IndexSet<H160> {
+    // newer orders at the start
+    orders.sort_by_key(|o| std::cmp::Reverse(o.metadata.creation_date));
+    let now = chrono::Utc::now();
+    let market_order_age = chrono::Duration::minutes(30);
+
+    let mut high_priority_tokens = IndexSet::new();
```

Comment:
nit, since it's all tokens

```suggestion
    let mut tokens_by_priority = IndexSet::new();
```